### PR TITLE
fix(sns): Don't let very old upgrade proposals block future upgrades

### PR DESCRIPTION
**Pull request by anchpop**
For those who don't know, upgrade proposals (e.g. UpgradeSnsToNextVersion) block all other upgrade actions while they're adopted until they're done executing. If a proposal were to get stuck, this would be a major problem, since the SNS would be un-upgradable. This obviously is a pretty bad scenario, so this PR makes it so only upgrade proposals decided within the last day can block other upgrades.

### CODE REVIEW

File: rs/sns/governance/src/governance.rs [hunk 1]
```diff
@@ -166,6 +166,10 @@ pub const UPGRADE_STEPS_INTERVAL_REFRESH_BACKOFF_SECONDS: u64 = 60 * 60; // 1 ho
 /// Past this duration, the lock will be automatically released.
 const UPGRADE_PERIODIC_TASK_LOCK_TIMEOUT_SECONDS: u64 = 600;
 
+/// Adopted-but-not-yet-executed upgrade proposals block other upgrade proposals from executing.
+/// But this is only true for proposals that are less than 1 day old, to prevent a stuck proposal from blocking all upgrades forever.
+const PROPOSAL_TOO_OLD_TO_BLOCK_OTHER_PROPOSALS: u64 = 60 * 60 * 24; // 1 day
```
Comment (aterga) [inline rs/sns/governance/src/governance.rs:171]:
```suggestion
const UPGRADE_PROPOSAL_BLOCK_EXPIRY_SECONDS: u64 = 60 * 60 * 24; // 1 day
```

File: rs/sns/governance/src/governance.rs [hunk 2]
```diff
@@ -2567,7 +2577,11 @@ impl Governance {
         proposal_id: Option<u64>,
     ) -> Result<(), GovernanceError> {
         let upgrade_proposals_in_progress = self.upgrade_proposals_in_progress();
-        if upgrade_proposals_in_progress != proposal_id.into_iter().collect() {
+        println!(
```
Comment (aterga) [inline rs/sns/governance/src/governance.rs:2580]: Is this intended to stay?

File: rs/sns/governance/src/governance/assorted_governance_tests.rs [hunk 3]
```diff
@@ -3330,6 +3330,75 @@ fn test_upgrade_proposals_blocked_by_pending_upgrade() {
...
+
+    let upgrade_action_id: u64 =
+        (&Action::UpgradeSnsControlledCanister(UpgradeSnsControlledCanister::default())).into();
+
+    let proposal_id = 1_u64;
+    let some_other_proposal_id = 99_u64;
+    let proposal = ProposalData {
+        action: upgrade_action_id,
+        id: Some(proposal_id.into()),
+        decided_timestamp_seconds: NativeEnvironment::DEFAULT_TEST_START_TIMESTAMP_SECONDS
```
Comment (aterga) [inline rs/sns/governance/src/governance/assorted_governance_tests.rs:3346]: Could you summarize in the doc string what this test is checking, please?

**Review by aterga — APPROVED** [review]

File: rs/sns/governance/src/governance/assorted_governance_tests.rs (hunk 3 above)
Comment (anchpop) [reply rs/sns/governance/src/governance/assorted_governance_tests.rs:3347]: Done!

File: rs/sns/governance/src/governance.rs (hunk 2 above)
Comment (anchpop) [reply rs/sns/governance/src/governance.rs]: Nope!

File: rs/sns/governance/src/governance.rs (hunk 1 above)
Comment (anchpop) [reply rs/sns/governance/src/governance.rs]: Done!
